src/server.c

```diff
@@ -6966,22 +6966,32 @@ int main(int argc, char **argv) {
             /* Either first or last argument - Should we read config from stdin? */
             if (argv[j][0] == '-' && argv[j][1] == '\0' && (j == 1 || j == argc-1)) {
                 config_from_stdin = 1;
+                j++;
             }
             /* All the other options are parsed and conceptually appended to the
              * configuration file. For instance --port 6380 will generate the
              * string "port 6380\n" to be parsed after the actual config file
              * and stdin input are parsed (if they exist). */
-            else if (argv[j][0] == '-' && argv[j][1] == '-') {
+            else {
                 /* Option name */
                 if (sdslen(options)) options = sdscat(options,"\n");
-                options = sdscat(options,argv[j]+2);
+                if (argv[j][0] == '-' && argv[j][1] == '-') {
+                    /* argv[j]+2 for removing the preceding `--` */
+                    options = sdscat(options,argv[j]+2);
+                } else {
+                    /* Invalid options, add them directly, and verify them in config.c */
+                    options = sdscat(options,argv[j]);
+                }
```

Comment:
this will make `src/redis-server --logfile --my--log--file loglevel warning` pass... 
i am thinking another way to raise the error ASAP

Reply:
wouldn't it be sufficient to add another condition to the original
`else if (argv[j][0] == '-' && argv[j][1] == '-') {` making sure not to enter if we don't already have at least one config value for the last config?

Reply:
yes. silly me, i just pushed a new commit (hard to take a name) and saw your comment